tests/apps/social/test_pipeline_user.py

```diff
@@ -0,0 +1,88 @@
+"""
+Test suite for pipeline user.
+"""
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.test import RequestFactory, TestCase, override_settings
+
+from social_core.exceptions import AuthAlreadyAssociated, AuthFailed
+from social_django.utils import load_backend, load_strategy
+
+from richie.apps.core.factories import UserFactory
+from richie.apps.social.pipeline.user import get_username
+
+
+class PipelineUserTestCase(TestCase):
+    """Tests for pipeline user."""
+
+    def setUp(self):
+        """Configure social-auth strategy."""
+        self.request_factory = RequestFactory()
+        self.request = self.request_factory.get("/", data={"x": "1"})
+        SessionMiddleware().process_request(self.request)
```

Comment:
Curious: 
Can you explain why you need data?
What is the middleware that you  need to run here?

Reply:
The `DjangoStrategy` use the session so we have to add it through the `SessionMiddleware`. Withoutit we have this error : 

```
____________________________________ PipelineUserTestCase.test_get_username_no_username_in_backend ____________________________________

self = <tests.apps.social.test_pipeline_user.PipelineUserTestCase testMethod=test_get_username_no_username_in_backend>

    def setUp(self):
        """Configure social-auth strategy."""
        self.request_factory = RequestFactory()
        self.request = self.request_factory.get("/", data={"x": "1"})
>       self.strategy = load_strategy(request=self.request)

tests/apps/social/test_pipeline_user.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.7/site-packages/social_django/utils.py:23: in load_strategy
    return get_strategy(STRATEGY, STORAGE, request)
/usr/local/lib/python3.7/site-packages/social_core/utils.py:278: in get_strategy
    return Strategy(Storage, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <social_django.strategy.DjangoStrategy object at 0x7fc598ac7250>, storage = <class 'social_django.models.DjangoStorage'>
request = <WSGIRequest: GET '/?x=1'>, tpl = None

    def __init__(self, storage, request=None, tpl=None):
        self.request = request
>       self.session = request.session if request else {}
E       AttributeError: 'WSGIRequest' object has no attribute 'session'

/usr/local/lib/python3.7/site-packages/social_django/strategy.py:39: AttributeError
```

Reply:
But data can be removed.